packages/forms/frontend/sirius-components-widget-reference/src/components/FilterableSortableList.tsx

```diff
@@ -0,0 +1,195 @@
+import { ServerContext } from '@eclipse-sirius/sirius-components-core';
+import { splitText } from '@eclipse-sirius/sirius-components-trees';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import { makeStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+import DragHandleIcon from '@material-ui/icons/DragHandle';
+import React, { useContext, useState } from 'react';
+import { FilterableSortableListProps } from './FilterableSortableList.types';
+import { ModelBrowserFilterBar } from './ModelBrowserFilterBar';
+
+const useStyles = makeStyles((theme) => ({
+  selectable: {
+    cursor: 'pointer',
+    '&:hover': {
+      stroke: '#66808a',
+      fill: '#66808a',
+      backgroundColor: 'var(--blue-lagoon-lighten-95)',
+    },
+  },
+  selected: {
+    cursor: 'pointer',
+    stroke: '#66808a',
+    fill: '#66808a',
+    backgroundColor: 'var(--blue-lagoon-lighten-90)',
+  },
+  selectedLabel: {
+    fontWeight: 'bold',
+  },
+  title: {
+    opacity: 0.6,
+    fontSize: 12,
+  },
+  borderStyle: {
+    border: '1px solid #0000008A',
+    height: 300,
+    overflow: 'auto',
+  },
+  highlight: {
+    backgroundColor: theme.palette.navigation.leftBackground,
+  },
+}));
+
+const highlightText = (label: string, textToHighlight: string, classes): JSX.Element => {
+  let itemLabel: JSX.Element;
+  const splitLabelWithTextToHighlight: string[] = splitText(label, textToHighlight);
+  if (
+    textToHighlight === null ||
+    textToHighlight === '' ||
+    (splitLabelWithTextToHighlight.length === 1 &&
+      splitLabelWithTextToHighlight[0].toLocaleLowerCase() !== label.toLocaleLowerCase())
+  ) {
+    itemLabel = <>{label}</>;
+  } else {
+    const languages: string[] = Array.from(navigator.languages);
+    itemLabel = (
+      <>
+        {splitLabelWithTextToHighlight.map((value, index) => {
+          const shouldHighlight = value.localeCompare(textToHighlight, languages, { sensitivity: 'base' }) === 0;
+          return (
+            <span
+              key={value + index}
+              data-testid={`${label}-${value}-${index}`}
+              className={shouldHighlight ? classes.highlight : ''}>
+              {value}
+            </span>
+          );
+        })}
+      </>
+    );
+  }
+  return itemLabel;
+};
+
+export const FilterableSortableList = ({
+  items,
+  setItems,
+  handleDragItemStart,
+  handleDragItemEnd,
+  handleDropNewItem,
+  onClick,
+  selectedItems,
+}: FilterableSortableListProps) => {
+  const classes = useStyles();
+  const { httpOrigin } = useContext(ServerContext);
+
+  const [filterBarText, setFilterBarText] = useState<string>('');
+  const [isHovered, setIsHovered] = useState<string>(null);
+  const [draggingItemId, setDraggingItemId] = useState<string>(null);
```

Comment:
Use one state

Reply:
Why use only one state ? For states that concerned not related variables, it seems to be a good practice to split in seperate state (https://legacy.reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables).
I find it more readable.

Reply:
done